genie-web/src/main/java/com/netflix/genie/web/services/impl/JobOwnerServiceImpl.java

```diff
@@ -0,0 +1,91 @@
+/*
+ *
+ *  Copyright 2016 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+package com.netflix.genie.web.services.impl;
+
+import com.netflix.genie.common.exceptions.GenieException;
+import com.netflix.genie.common.exceptions.GenieNotFoundException;
+import com.netflix.genie.web.services.AgentRoutingService;
+import com.netflix.genie.web.services.JobOwnerService;
+import com.netflix.genie.web.services.JobPersistenceService;
+import com.netflix.genie.web.services.JobSearchService;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Service;
+
+import javax.validation.constraints.NotBlank;
+
+/**
+ * Service to get information about the server owning a job.
+ * Job can be a v3 job which runs locally on a genie node. This node is the job owner.
+ * Job can be a v4 job running remotely on a agent. The genie node which has the
+ * connection to this agent is the owner.
+ *
+ * @author standon
+ * @since 4.0.0
+ */
+@Slf4j
+@Service
+@Lazy
+public class JobOwnerServiceImpl implements JobOwnerService {
+
+    private final AgentRoutingService agentRoutingService;
+    private final JobSearchService jobSearchService;
+    private final JobPersistenceService jobPersistenceService;
+
+    /**
+     * Constructor.
+     *
+     * @param agentRoutingService       Service to routing information for jobs running on agent(v4 jobs)
+     * @param jobPersistenceService     Job persistence service
+     * @param jobSearchService          Service to search v3 jobs
+     */
+    public JobOwnerServiceImpl(
+        final AgentRoutingService agentRoutingService,
+        final JobSearchService jobSearchService,
+        final JobPersistenceService jobPersistenceService
+
+        ) {
+        this.agentRoutingService = agentRoutingService;
+        this.jobSearchService = jobSearchService;
+        this.jobPersistenceService = jobPersistenceService;
+    }
+
+    @Override
```

Comment:
the `@Override` should be directly above the code not the comment. Also no need to duplicate the comment as it should just inherit the one from the interface using `{@inherit}` and then anything additional you can put above or below that.